Tools/clean.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```

Comment:
Why do we have a bash file?

Reply:
Specifically in MacOS, autorest would generate a file that could not be deleted within the /jumpcloud.sdk.*/bin or /jumpcloud.sdk.*/obj directory. If that was the case, you'd have to go through manually and delete this file / directory. This clean.sh file just sets permissions for everyone on those two directories/ removes contents so the build-module.ps1 can execute. 

The first action of the build-module.ps1 file is to delete the /jumpcloud.sdk.*/bin or /jumpcloud.sdk.*/obj directories. This just automates this for those of us building locally on MacOS. I added a check on BuildAutoRest.ps1 to prevent this from running in other operating systems.

Reply:
Can we write this script in pwsh? Seeing as that all the scripts in this repo are pwsh.